src/recon_buildblock/KappaComputation.cxx

```diff
@@ -0,0 +1,264 @@
+/*
+    Copyright (C) 2020 University College London
+    This file is part of STIR.
+
+    This file is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This file is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    See STIR/LICENSE.txt for details
+*/
+
+/*!
+  \file
+  \ingroup recon_buildblock
+  \brief Declaration of class stir::KappaComputation
+
+  \author Robert Twyman
+  \author Kris Thielemans
+*/
+
+#include "stir/recon_buildblock/KappaComputation.h"
+#include "stir/info.h"
+#include "stir/modelling/ParametricDiscretisedDensity.h"
+#include "stir/IO/OutputFileFormat.h"
+#include "stir/IO/read_from_file.h"
+#include "stir/is_null_ptr.h"
+//#include "stir/recon_buildblock/GeneralisedObjectiveFunction.h"
+//#include "stir/recon_buildblock/GeneralisedPrior.h"
+#include "stir/recon_buildblock/PoissonLogLikelihoodWithLinearModelForMeanAndProjData.h"
+#include "stir/unique_ptr.h"
+
+
+START_NAMESPACE_STIR
+
+template <typename TargetT>
+KappaComputation<TargetT>::
+KappaComputation()
+{
+  this->set_defaults();
+}
+
+template <typename TargetT>
+void
+KappaComputation<TargetT>::
+set_defaults()
+{
+  objective_function_sptr.reset(new PoissonLogLikelihoodWithLinearModelForMeanAndProjData<TargetT>);
```

Comment:
@KrisThielemans This is currently the problem I have when trying to allow other templates including 
```
template class KappaComputation<ParametricVoxelsOnCartesianGrid >;
```
Ideally `objective_function_sptr` should be set to different types of objective_function depending on what `TargetT` (template) is, but I dont know how to do this and because I can I think I am miss understanding the usage of abstract classes.

Reply:
I think the solution is not to set a default for `objective_function_sptr`. What input data would the default have anyway?

Reply:
Removing this default solved a lot of problems.